src/app/labeling/qgsmaptoollabel.cpp

```diff
@@ -678,6 +678,101 @@ bool QgsMapToolLabel::currentLabelDataDefinedPosition( double &x, bool &xSuccess
   return true;
 }
 
+bool QgsMapToolLabel::currentLabelDataDefinedPosition( double &x, bool &xSuccess, double &y, bool &ySuccess, int &xCol, int &yCol, int &pointCol ) const
+{
+  QgsVectorLayer *vlayer = mCurrentLabel.layer;
+  QgsFeatureId featureId = mCurrentLabel.pos.featureId;
+
+  xSuccess = false;
+  ySuccess = false;
+
+  if ( !vlayer )
+  {
+    return false;
+  }
+
+  if ( mCurrentLabel.pos.isDiagram )
+  {
+    if ( !diagramMoveable( vlayer, xCol, yCol ) )
+    {
+      return false;
+    }
+  }
+  else if ( !labelMoveable( vlayer, mCurrentLabel.settings, xCol, yCol, pointCol ) )
+  {
+    return false;
+  }
+
+  QgsFeature f;
+  if ( !vlayer->getFeatures( QgsFeatureRequest().setFilterFid( featureId ).setFlags( QgsFeatureRequest::NoGeometry ) ).nextFeature( f ) )
+  {
+    return false;
+  }
+
+  if ( !mCurrentLabel.pos.isUnplaced )
+  {
+    QgsAttributes attributes = f.attributes();
+
+    if ( mCurrentLabel.settings.dataDefinedProperties().isActive( QgsPalLayerSettings::PositionPoint ) )
+    {
+      if ( !attributes.at( pointCol ).isNull() )
+      {
+        QVariant pointAsVariant = attributes.at( pointCol );
+        if ( pointAsVariant.canConvert<QgsGeometry>() )
+        {
+          QgsGeometry geometryPoint = pointAsVariant.value<QgsGeometry>();
+          const QgsPoint *point  = qgsgeometry_cast<QgsPoint *>( geometryPoint.constGet() );
+
+          x = point->x();
+          y = point->y();
+
+          xSuccess = true;
+          ySuccess = true;
+        }
+        else if ( !pointAsVariant.toByteArray().isEmpty() )
```

Comment:
Where are bytearray wkb points produced?

Reply:
I'm guessing it's to support (Read only) fields which are already created on the database as wkb.

Personally I don't think this a safe approach -- we're feeding any random binary field into the fromWkb method, without any safeguard that it is actually wkb content. 

I think we could drop this handling here and instead just get users to use an expression based property with a `geom_from_wkb( "wkb_field")` expression if they need this functionality.